modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/message/MetastorageNodesResponse.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.metastorage.message;
+
+import java.io.Serializable;
+import java.util.Collection;
+import org.apache.ignite.network.NetworkMessage;
+import org.apache.ignite.network.annotations.Transferable;
+
+/**
+ * Response for the {@link MetastorageNodesRequest}.
+ */
+@Transferable(MetastorageMessages.METASTORAGE_NODES_RESPONSE)
+public interface MetastorageNodesResponse extends NetworkMessage, Serializable {
```

Comment:
Should NetworkMessage by itself be serializable?

Reply:
the problem is that `Serializable` is currently a hack. Not all `NetworkMessage`s need to be `Serializable`, even now some internal messages already use the Direct Marshaller.